### ISSUE TITLE
C++ instance name highlighted as type

### ISSUE THREAD

**Issue by elextr** [geany/geany#892]
In Geany the name `instance` is highlighted as a type

```C++
class c {
    int i;
public:
    c( int j ):i(j){}
    int f(){ return i; }
} instance{1};

c another{1};
```

Note github has it correct.

**Comment by b4n** [comment]
It's the initialization (`{1}`) that confuses the C parser.

**Comment by elextr** [comment]
Probably, but this is C++ ( hint, the `class` keyword :).  It accepts {} initialization on `another` though.

**Comment by b4n** [comment]
> It accepts {} initialization on `another` though.

Does it?  doesn't seem to show for me.

**Comment by elextr** [comment]
Oh, I was only looking at the highlighting, yeah it doesn't make a symbol for `another`.

Part of the general C++11 blindness then, thought it might be something specific, thats why I made the issue.

**Comment by techee** [comment]
The new parser fixes the `instance` highlighting and makes tag for `another` but it fails to generate the variable tag for `instance` (again, the `{1}` seems to confuse it). The world would be a boring place without some problems with the c++ parser I guess :-).

**Comment by elextr** [comment]
> The world would be a boring place without some problems with the c++ parser I guess :-).

Don't worry, I have heaps more to keep you "amused" into the future.

**Comment by elextr** [comment]
> (again, the {1} seems to confuse it)

yeah, a tag is made for `instance` if `()` is used instead of `{}`.

**Comment by techee** [comment]
> Don't worry, I have heaps more to keep you "amused" into the future.

The point of the whole ctags sync stuff is that someone else outside Geany will get "amused" and we'll just be those complaining about the problems :-)

**Comment by techee** [comment]
Created a bug report here: https://github.com/universal-ctags/ctags/issues/3280
